src/main/java/ecdar/backend/BackendHelper.java

```diff
@@ -67,12 +69,78 @@ public static void stopQueries() {
     /**
      * Generates a reachability query based on the given location and component
      *
-     * @param location  The location which should be checked for reachability
-     * @param component The component where the location belong to / are placed
+     * @param endLocation  The location which should be checked for reachability
      * @return A reachability query string
      */
-    public static String getLocationReachableQuery(final Location location, final Component component) {
-        return component.getName() + "." + location.getId();
+    public static String getLocationReachableQuery(final Location endLocation, final Component component) {
+        var stringBuilder = new StringBuilder();
+
+        // append simulation query (currently only supports parallel composition)
+        stringBuilder.append(getSimulationQueryString());
+
+        // append start location here TODO
+
+        // append end state
+        stringBuilder.append(getEndStateString(component.getName(), endLocation.getId()));
+
+        // append clocks
+        stringBuilder.append("(");
+        // append clock here TODO
+        stringBuilder.append(")");
+
+        //  return example: m1||M2->[L1,L4](y<3);[L2, L7](y<2)
+        return stringBuilder.toString();
+    }
+
+    private static String getSimulationQueryString() {
+        var stringBuilder = new StringBuilder();
+
+        var appendComponentWithSeparator = false;
+        for (var componentName:ListOfComponents) {
+            if (appendComponentWithSeparator){
+                stringBuilder.append("||" + componentName);
+            }
+            else {
+                stringBuilder.append(componentName);
+            }
+            if (!appendComponentWithSeparator) {
+                appendComponentWithSeparator = true;
+            }
+        }
+        return stringBuilder.toString();
+    }
```

Comment:
Tager I kun højde for parallel composition af to component?
Hvad med conjunction og quotient?

Reply:
Ja, det er kun den der fremgår af den grammatik, I sendte til os, og vi kan også kun lave simulation på parallel compositions lige nu.

Reply:
Altså, det fungere også både for conjunction og quotient, men hvis I ikke kommer til at lave det, kan i bare resolve denne conversation